### PR TITLE
HybridCompute stable release version 2022-12-27

### PR DESCRIPTION
**Pull request by yaotongms**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs [hunk 1]
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
...
+        /// <summary> Specifies the AD fully qualified display name. </summary>
+        public string AdFqdn { get; }
+        /// <summary> Specifies the DNS fully qualified display name. </summary>
+        public string DnsFqdn { get; }
+        /// <summary> The resource id of the private link scope this machine is assigned to, if any. </summary>
+        public string PrivateLinkScopeResourceId { get; set; }
+        /// <summary> The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any. </summary>
+        public string ParentClusterResourceId { get; set; }
+        /// <summary> Specifies whether any MS SQL instance is discovered on the machine. </summary>
+        public string MssqlDiscovered { get; set; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:167]: Should that be `MsSqlDiscovered`?
```suggestion
        public string MsSqlDiscovered { get; set; }
```

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs [hunk 2]
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
...
+        /// <summary> Specifies the Operating System product SKU. </summary>
+        public string OSSku { get; }
+        /// <summary> Specifies the Windows domain name. </summary>
+        public string DomainName { get; }
+        /// <summary> Specifies the AD fully qualified display name. </summary>
+        public string AdFqdn { get; }
+        /// <summary> Specifies the DNS fully qualified display name. </summary>
+        public string DnsFqdn { get; }
+        /// <summary> The resource id of the private link scope this machine is assigned to, if any. </summary>
+        public string PrivateLinkScopeResourceId { get; set; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:163]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

Comment (archerzz) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:163]: Same question for `ParentClusterResourceId`.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs [hunk 3]
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
...
+        /// <summary> Public Key that the client provides to be used during initial resource onboarding. </summary>
+        public string ClientPublicKey { get; set; }
+        /// <summary> The Operating System running on the hybrid machine. </summary>
+        public string OSName { get; }
+        /// <summary> The version of Operating System running on the hybrid machine. </summary>
+        public string OSVersion { get; }
+        /// <summary> The type of Operating System (windows/linux). </summary>
+        public string OSType { get; set; }
+        /// <summary> Specifies the Arc Machine's unique SMBIOS ID. </summary>
+        public string VmUuid { get; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:151]: Can we change the type to [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0)? See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs [hunk 4]
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
...
+        /// <summary> The status of the hybrid machine agent. </summary>
+        public HybridComputeStatusType? Status { get; }
+        /// <summary> The time of the last status change. </summary>
+        public DateTimeOffset? LastStatusChange { get; }
+        /// <summary> Details about the error state. </summary>
+        public IReadOnlyList<ResponseError> ErrorDetails { get; }
+        /// <summary> The hybrid machine agent full version. </summary>
+        public string AgentVersion { get; }
+        /// <summary> Specifies the hybrid machine unique ID. </summary>
+        public string VmId { get; set; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:137]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            EnableAutomaticUpgrade = enableAutomaticUpgrade;
+            LastAttemptTimestamp = lastAttemptTimestamp;
+            LastAttemptStatus = lastAttemptStatus;
+            LastAttemptMessage = lastAttemptMessage;
+        }
+
+        /// <summary> Specifies the version info w.r.t AgentUpgrade for the machine. </summary>
+        public string DesiredVersion { get; set; }
+        /// <summary> The correlation ID passed in from RSM per upgrade. </summary>
+        public string CorrelationId { get; set; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs:38]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        /// <summary> Specifies if RSM should try to upgrade this machine. </summary>
+        public bool? EnableAutomaticUpgrade { get; set; }
+        /// <summary> Timestamp of last upgrade attempt. </summary>
+        public string LastAttemptTimestamp { get; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs:42]: Could this be a type of `DateTimeOffset`? See an example [here](https://github.com/Azure/azure-sdk-for-net/blob/e4fce1607d65d709b19c81cc78969cd08b804244/sdk/hdinsight/Azure.ResourceManager.HDInsight/src/autorest.md?plain=1#L146).

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs [hunk 7]
```diff
@@ -19,7 +19,23 @@ public HybridComputeMachinePatch()
...
+        /// <summary> Specifies the cloud provider (Azure/AWS/GCP...). </summary>
+        public string CloudMetadataProvider
+        {
+            get => CloudMetadata is null ? default : CloudMetadata.Provider;
+        }
+
+        /// <summary> The info of the machine w.r.t Agent Upgrade. </summary>
+        public AgentUpgrade AgentUpgrade { get; set; }
+        /// <summary> The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any. </summary>
+        public string ParentClusterResourceId { get; set; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs:37]: Same question regarding the format of the ID

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/MachineAssessPatchesResult.cs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+            StartedBy = startedBy;
+            PatchServiceUsed = patchServiceUsed;
+            OSType = osType;
+            ErrorDetails = errorDetails;
+        }
+
+        /// <summary> The overall success or failure status of the operation. It remains "InProgress" until the operation completes. At that point it will become "Unknown", "Failed", "Succeeded", or "CompletedWithWarnings.". </summary>
+        public PatchOperationStatus? Status { get; }
+        /// <summary> The activity ID of the operation that produced this result. </summary>
+        public string AssessmentActivityId { get; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/MachineAssessPatchesResult.cs:49]: same question for the format of ID

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs (hunk 2 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:163]: @archerzz I got an error opening the example links although being a member of the microsoft org. Wonder if you can send out another link or screenshot for the ResourceIdentifier example.

image

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs (hunk 3 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs]: changed the type of VmUuid to Guid.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs (hunk 4 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs]: changed the type of VmId to Guid.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs (hunk 5 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs]: changed the type of CorrelationId to Guid.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs (hunk 6 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs]: changed the type of LastAttemptTimestamp to DateTimeOffset.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs (hunk 7 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs:37]: changed the type of ParentClusterResourceId to Guid.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/MachineAssessPatchesResult.cs (hunk 8 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/MachineAssessPatchesResult.cs]: changed the type of AssessmentActivityId to Guid.

**Comment by yaotongms** [comment]
Test recordings uploaded to: https://github.com/Azure/azure-sdk-assets/tree/net/hybridcompute/Azure.ResourceManager.HybridCompute_8d2d4b8453

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs (hunk 7 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs]: fixed in the latest commit

**Pull request closed without merging**